### PR TITLE
Registry replacer pruning: Consider copy --from and allow alias replacement

### PR DESCRIPTION
**Pull request by alvaroaleman**

### CODE REVIEW

File: cmd/registry-replacer/main.go [hunk 1]
```diff
@@ -143,7 +143,11 @@ func replacer(
 		}
 
 		getter := githubFileGetterFactory(info.Org, info.Repo, info.Branch)
-		allSourceImages := sets.String{}
+		allReplacementCandidates := sets.String{}
+
+		// We have to skip pruning if we only get empty dockerfiles because it might mean
+		// that we do not have the appropriate permissions.
+		var hasNonEmptyDockerfile bool
```
Comment (stevekuznetsov) [inline cmd/registry-replacer/main.go:150]: Let's make this discoverable since it will mean that we still have non-conformant configs

Comment (alvaroaleman) [reply cmd/registry-replacer/main.go:150]: Done

**Comment by alvaroaleman** [comment]
> So I vaguely remember Clayton saying that the image replacement does not actually work when you want to replace the as <name> and target the name -- did we confirm that users which configured it to do so, actually got what they wanted?

It does work if what is aliased in the `as` is not getting replaced already. The code this is using now is copied from what their replacer does

**Pull request merged**